src/components/SubRequirement.vue

```diff
@@ -142,7 +204,76 @@ export default Vue.extend({
       this.showFulfillmentOptionsDropdown = false;
       this.$emit('changeToggleableRequirementChoice', this.subReq.id, option);
     },
-  },
+    generateSubReqCoursesNotTakenArray(){
+      // Reset subReqCoursesNotTakenArray
+      this.subReqCoursesNotTakenArray = [];
+
+      // Depending on fulfilledBy, subReqCourses is accessed differently from subReq
+      let subReqCourses;
+      if (this.subReq.requirement.fulfilledBy === 'toggleable') {
+        const fulfillmentOption = this.toggleableRequirementChoice || this.selectedFulfillmentOption;
+        subReqCourses = this.subReq.requirement.fulfillmentOptions[fulfillmentOption].courses;
+      } else if (this.subReq.requirement.fulfilledBy === 'self-check') {
+        subReqCourses = [];
+      } else { // fulfilledBy courses or credits
+        subReqCourses = this.subReq.requirement.courses;
+      }
+      const mostRecentRosters = this.rostersFromLastTwoYears;
+      let filteredSubReqRosters;
+      // Iterate over each course slot for the subReq
+      subReqCourses.forEach((subReqCourseRosterObject:EligibleCourses) => {
+        // Filter subreq roster object keys with the mostRecentRosters
+        filteredSubReqRosters = Object.keys(subReqCourseRosterObject).filter(subReqRoster => mostRecentRosters.indexOf(subReqRoster) !== -1).reverse();
+
+        const crseInfoObjects: CrseInfo[] = []; // List of crseInfoObjects {roster: <roster>, crseIds: crseId[]} []
+        let seenCrseIds: number[] = []; // So we don't have duplicates
```

Comment:
if you want to eliminate duplicates, use `Set`. It's more efficient, and provides better API like `set.has(...)`